### PR TITLE
Add Schmitt Trigger Block with Uncertainty Support and Optional Interpolation

### PR DESCRIPTION
**Pull request by RalphSteinhagen**
This PR introduces a new digitial `SchmittTrigger` block with configurable threshold and offset, as well as sub-sample edge timing interpolation options.

### Features

- **Edge Detection**: detects rising and falling edges based on user-defined `threshold` and `offset` values.
- **Uncertainty Support**: Propagates uncertainties and provides precise edge timing with associated errors.
- **Interpolation Methods**: Supports multiple methods for sub-sample precision:
  - `NO_INTERPOLATION`: basic detection without interpolation.
  - `BASIC_LINEAR_INTERPOLATION`: simple linear interpolation between adjacent samples.
  - `LINEAR_INTERPOLATION`: linear regression over the samples within the threshold window.
  - `POLYNOMIAL_INTERPOLATION`: (TODO) Placeholder for future implementation using Savitzky–Golay filters.
- **Trigger Generation**: Generates customizable triggers (`trigger_name_rising_edge`, `trigger_name_falling_edge`)

image

## FIXME and TODOs

- **FIXME**: In the `processBulk` method, there's a commented-out section related to input sample buffering and history. This needs to be revisited to ensure proper operation, especially when using interpolation methods that require a history buffer. The error seems to be unrelated to the underlying algorithm and probably also chosen block implementation (tbc.). To be followed up in the GR4 if that is the case.

- **TODO**: The `POLYNOMIAL_INTERPOLATION` method has not yet been implemented and requires the pending `Tensor<T>` and SVD implementations. This will be addressed in a future update.

### CODE REVIEW

**Review by wirew0rm — COMMENTED** [review]
Tried to look into it but got stuck trying to get it to compile and it was not clear to me what the right direction to fix this would be.

But overall looks very nice, I suspect this was something that was changed last minute and missed a few parts of the code.

EDIT: after writing this I gave it one more quick try with more understanding and got it working. I also added fixup commits for the code formatting.

File: algorithm/include/gnuradio-4.0/algorithm/SchmittTrigger.hpp [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    value_t _lowerThreshold;
+
+    bool _lastState{false}; // true if above upper threshold
+
+    HistoryBuffer<T, interpolationWindow> _historyBuffer;
+
+    // edge timing variables
+    EdgeDetection lastEdge = EdgeDetection::NONE;
+    EdgeType      lastEdgeIdx{1};    /// relative index [-EdgeType|_MAX,1[ w.r.t. current sample of the last detected edge
+    EdgeType      lastEdgeOffset{0}; /// relative sub-sample offset [0, 1[ w.r.t. current sample of the last detected edge
```
[the remark this answers is not in the document]
Comment (RalphSteinhagen) [reply algorithm/include/gnuradio-4.0/algorithm/SchmittTrigger.hpp]: The thing is that -- normally it's:
 * `float` -> `float`
 * `double` -> `double`
 * `UncertainValue<float>` -> `UncertainValue<float>`
 * `UncertainValue<double>` -> `UncertainValue<double>`
 
 The integer types are a bit special because while the vertical values are discrete, the time-offsets aren't, thus my choice to default to `float` or `UncertainValue<float>` for them. Was also temporarily contemplating keeping them all as `float` or `UncertainValue<float>` but this would complicate the downstream logic probably.
 
 My goal was first to have a viable working solution and to see how it works out in the field.

**Comment by RalphSteinhagen** [comment]
fixed tag indexing problem and upgraded unit-tests to cover some more edge cases:

image

**Review by wirew0rm — APPROVED** [review]
Looks really nice, thanks for the nice implementation, documentation, tests and issue follow-up. :+1:
